app/models/links.rb

```diff
@@ -0,0 +1,69 @@
+class Links
+  # Links available for indexed gems
+  LINKS = {
+    'home'     => 'homepage_uri',
+    'code'     => 'source_code_uri',
+    'docs'     => 'documentation_uri',
+    'wiki'     => 'wiki_uri',
+    'mail'     => 'mailing_list_uri',
+    'bugs'     => 'bug_tracker_uri',
+    'download' => 'download_uri'
+  }.freeze
+
+  # Links available for non-indexed gems
+  NON_INDEXED_LINKS = {
+    'docs'     => 'documentation_uri'
+  }.freeze
+
+  attr_accessor :rubygem, :version, :linkset
+
+  def initialize(rubygem = nil, version = nil)
+    self.rubygem = rubygem
+    self.version = version
+    self.linkset = rubygem.linkset
+  end
+
+  def links
+    version.indexed ? LINKS : NON_INDEXED_LINKS
+  end
+
+  delegate :keys, to: :links
+
+  def each
+    return enum_for(:each) unless block_given?
+    links.each do |short, long|
+      value = send(long)
+      yield short, value if value
+    end
+  end
+
+  # documentation url:
+  # if metadata has it defined, use that
+  # or if linksets has it defined, use that
+  # else, generate one from gem name and version number
+  def documentation_uri
+    (version.metadata && version.metadata["documentation_uri"].presence) ||
+      (linkset && linkset.docs.presence) ||
+      "http://www.rubydoc.info/gems/#{rubygem.name}/#{version.number}"
+  end
+
+  # technically this is a path
+  def download_uri
+    "/downloads/#{version.full_name}.gem" if version.indexed
+  end
+
+  def badge_uri
```

Comment:
Oh gotcha! :+1: 

I think I am going to push for removal badge fury link all together, although not in this PR.